workitem/expression_compiler_blackbox_test.go

```diff
@@ -16,7 +16,7 @@ func TestField(t *testing.T) {
 	defJoins := workitem.DefaultTableJoins()
 
 	wiTbl := workitem.WorkItemStorage{}.TableName()
-	expect(t, c.Equals(c.Field("foo.bar"), c.Literal(23)), `(`+workitem.Column(wiTbl, "fields")+` @> '{"foo.bar" : 23}')`, []interface{}{}, nil)
+	expect(t, c.Equals(c.Field("foo_bar"), c.Literal(23)), `(`+workitem.Column(wiTbl, "fields")+` @> '{"foo_bar" : 23}')`, []interface{}{}, nil)
```

Comment:
Why do you need to make this change, when fields that are not present in the NewToOld and OldToNew maps are ignored anyways? `foo.bar` should not be replaced IMHO because it also tests the expression compilers logic, that field names can have a `.` inside and they don't necessarily mean something.

Reply:
This change isn't directly related to the workitem conversion. We need this change because the new fields are of the form `system_foo`, `foo_bar`, etc. Without this change the test would fail.

The expression compiler also needs to know that the new fields are of the form `system_*
See this piece of code
```go
	// Check if the field name contains an underscore
	if strings.Contains(fieldName, "_") {
		// leave field untouched
		return fieldName, true
	}
```

Expression compiler has been updated and so the test also need to be modified accordingly.

Reply:
Can't we just have these two tests, one existing and one that didn't exist?

```go
expect(t, c.Equals(c.Field("foo.bar"), c.Literal(23)), `(`+workitem.Column(wiTbl, "fields")+` @> '{"foo.bar" : 23}')`, []interface{}{}, nil)
expect(t, c.Equals(c.Field("foo_bar"), c.Literal(23)), `(`+workitem.Column(wiTbl, "fields")+` @> '{"foo_bar" : 23}')`, []interface{}{}, nil)
```

I'm still not convinced why we don't need `foo.bar`. Do you want to completely get rid of any field with a `.` inside?

Reply:
```go
expect(t, c.Equals(c.Field("foo.bar"), c.Literal(23)), `(`+workitem.Column(wiTbl, "fields")+` @> '{"foo.bar" : 23}')`, []interface{}{}, nil)
```
would fail because `foo.bar` won't be considered as a `field` but it would be considered as a direct attribute of `workitem`.
So for `foo.bar:xxx`, it would be `workitem.foo.bar = xxx` and for 
`foo_bar:xxx` is would be `workitem.fields <@ {"foo_bar": xxx}`

Reply:
I've added a test to demonstate this https://github.com/fabric8-services/fabric8-wit/pull/2333/commits/97d38cbddb5c7a5ab1f2310dbb48d5b9e628f025

Reply:
> I've added a test to demonstate this [97d38cb](https://github.com/fabric8-services/fabric8-wit/commit/97d38cbddb5c7a5ab1f2310dbb48d5b9e628f025)

@jarifibrahim thank you for doing this. This adds so much documentation value to me!!